targets/TARGET_STM/TARGET_STM32F4/serial_device.c

```diff
@@ -28,242 +28,29 @@
  *******************************************************************************
  */
 
-#include "mbed_assert.h"
-#include "serial_api.h"
-#include "serial_api_hal.h"
-
 #if DEVICE_SERIAL
 
-#include "cmsis.h"
-#include "pinmap.h"
-#include <string.h>
-#include "PeripheralPins.h"
-#include "mbed_error.h"
+#include "serial_api_hal.h"
 
-#define UART_NUM (10)
-static uint32_t serial_irq_ids[UART_NUM] = {0};
+#if defined (TARGET_STM32F401xC) || defined (TARGET_STM32F401xE) || defined (TARGET_STM32F410xB) || defined (TARGET_STM32F411xE)
+    #define UART_NUM (3)
+#elif defined (TARGET_STM32F412xG)
+    #define UART_NUM (4)
+#elif defined (TARGET_STM32F407xG) || defined (TARGET_STM32F446xE)
+    #define UART_NUM (6)
+#elif defined (TARGET_STM32F429xI) || defined (TARGET_STM32F439xI) || defined (TARGET_STM32F437xG) || defined (TARGET_STM32F469xI)
+    #define UART_NUM (8)
+#elif defined (TARGET_STM32F413xH)
```

Comment:
maybe merge this #elif and the #else below ?

Reply:
done